### ISSUE TITLE
Sync Conflict Notification (in a few cases)

### ISSUE THREAD

**Issue by DerekCaelin** [techmatters/terraso-mobile-client#2324]
In the case that there is an issue while syncing, display a notification. 

Sync errors can occur when the user client `pushes` data and 
- A site was deleted or removed from a project
- The user was removed from a project
- Depth requirements for a project changed

We will also display this when a `pull` occurs and a user could be looking at content to which they no longer have access.

## Design
[Figma](https://www.figma.com/design/KMJbhjVyrrLFHj6jGcX9lN/LandPKS-Design-Prototype-working-file?node-id=10659-8623&node-type=frame&t=GgmGoTssg8eIF1h9-0)

<img width="392" alt="image" src="https://github.com/user-attachments/assets/9146beee-fd50-488c-9b46-c1dde7c34439">

## Acceptance Criteria
- [x] For this story: If the user client **pulls** and the user is looking at something they can no longer access, display the dialog in the cases that are already implemented: removing a site or site note on the "edit site note" screen, removing a site on the texture screen or one of the site tabs
- [x] Link to: https://landpks.terraso.org/help/project-conflict/

### Non-goals
- Display the dialog for all **pull** cases when user is looking at a site or project they can no longer access (will be covered in [#2290](https://github.com/techmatters/terraso-mobile-client/issues/2290))
- If the user client **pushes** data and 
   - A site was deleted or removed from a project
   - The user was removed from a project
   - Depth requirements for a project changed
....then display the dialog (will be covered in [#2293](https://github.com/techmatters/terraso-mobile-client/issues/2293))

**Comment by knipec** [comment]
Should be done after #2279 
Could be done after or alongside #2293

**Comment by tm-ruxandra** [comment]
This will be testable when #2293 is out in a build as well.

**Comment by CourtneyLee333** [comment]
I'm assuming this will be in v217.

**Comment by DerekCaelin** [comment]
@tm-ruxandra this notification appears when expected and passes the design review. Please update the "support" link to go here: https://landpks.terraso.org/help/synchronization-conflict/

**Comment by tm-ruxandra** [comment]
@DerekCaelin This fix is now in review, will notify you when it can be re-tested.

**Comment by DerekCaelin** [comment]
@tm-ruxandra I assume this is testable in 221?

**Comment by tm-ruxandra** [comment]
@DerekCaelin yes

**Comment by DerekCaelin** [comment]
Beautiful.
